lib/buffer.js

```diff
@@ -151,30 +150,20 @@ function fromObject(obj) {
   }
 
   if (obj.buffer instanceof ArrayBuffer || obj.length) {
```

Comment:
Not sure about performance impact though. That said, if our tests didn't catch this case, may I ask you to update the test with empty array case?